tensorflow/core/kernels/mkl_fused_batch_norm_op.cc

```diff
@@ -143,83 +179,152 @@ class MklFusedBatchNormFwdPrimitive : public MklPrimitive {
+               *context_.dst_mem }});
       }
-    } else if (context_.flags & use_global_stats) {
-      if ((context_.flags & use_scale_shift) && mkldnn_use_scaleshift) {
+      context_.bn_fwd.reset(new batch_normalization_forward(*context_.fwd_pd));
+#else
+      if ((context_.flags & (int)BN_FLAGS::use_scale_shift) &&
+          (int)BN_FLAGS::use_scale_shift) {
         context_.bn_fwd.reset(new batch_normalization_forward(
-            fwd_pd, *context_.src_mem, (const primitive::at)*context_.mean_mem,
+            *context_.fwd_pd, *context_.src_mem,
+            (const primitive::at)*context_.mean_mem,
             (const primitive::at)*context_.variance_mem, *context_.weights_mem,
             *context_.dst_mem));
       } else {
         context_.bn_fwd.reset(new batch_normalization_forward(
-            fwd_pd, *context_.src_mem, (const primitive::at)*context_.mean_mem,
+            *context_.fwd_pd, *context_.src_mem,
+            (const primitive::at)*context_.mean_mem,
             (const primitive::at)*context_.variance_mem, *context_.dst_mem));
       }
+#endif  // ENABLE_MKLDNN_V1
     } else {
-      if ((context_.flags & use_scale_shift) && mkldnn_use_scaleshift) {
+#ifdef ENABLE_MKLDNN_V1
+      if ((context_.flags & (int)BN_FLAGS::use_scale_shift) &&
+          (int)BN_FLAGS::use_scale_shift) {
+        context_.net_args.push_back(
+            {{MKLDNN_ARG_SRC, *context_.src_mem},
+             {MKLDNN_ARG_WEIGHTS, *context_.weights_mem},
+             {MKLDNN_ARG_DST, *context_.dst_mem},
+             {MKLDNN_ARG_MEAN, *context_.mean_mem},
+             { MKLDNN_ARG_VARIANCE,
+               *context_.variance_mem }});
+      } else {
+        context_.net_args.push_back({{MKLDNN_ARG_SRC, *context_.src_mem},
+                                     {MKLDNN_ARG_DST, *context_.dst_mem},
+                                     {MKLDNN_ARG_MEAN, *context_.mean_mem},
+                                     { MKLDNN_ARG_VARIANCE,
+                                       *context_.variance_mem }});
+      }
+      context_.bn_fwd.reset(new batch_normalization_forward(*context_.fwd_pd));
+#else
+      if ((context_.flags & (int)BN_FLAGS::use_scale_shift) &&
+          (int)BN_FLAGS::use_scale_shift) {
         context_.bn_fwd.reset(new batch_normalization_forward(
-            fwd_pd, *context_.src_mem, *context_.weights_mem, *context_.dst_mem,
-            *context_.mean_mem, *context_.variance_mem));
+            *context_.fwd_pd, *context_.src_mem, *context_.weights_mem,
+            *context_.dst_mem, *context_.mean_mem, *context_.variance_mem));
       } else {
         context_.bn_fwd.reset(new batch_normalization_forward(
-            fwd_pd, *context_.src_mem, *context_.dst_mem, *context_.mean_mem,
-            *context_.variance_mem));
+            *context_.fwd_pd, *context_.src_mem, *context_.dst_mem,
+            *context_.mean_mem, *context_.variance_mem));
       }
+#endif  // ENABLE_MKLDNN_V1
```

Comment:
I just realized that for the V1 case, the orders of the parameters in the set doesn't matter since the container is `unordered_map`. Then we can simplify it even further + use `emplace_back` instead of `push_back`:
```c++
#ifdef ENABLE_MKLDNN_V1
  std::unordered_map<int, memory> net_args_map = {
      {MKLDNN_ARG_SRC, *context_.src_mem},
      {MKLDNN_ARG_DST, *context_.dst_mem}};
  if (fwdParams.training || IS_SET(use_global_stats)) {
    net_args_map.insert({{MKLDNN_ARG_MEAN, *context_.mean_mem},
                         {MKLDNN_ARG_VARIANCE, *context_.variance_mem}}); 
  }
  if (IS_SET(use_scale_shift) {
    net_args_map.insert({{MKLDNN_ARG_WEIGHTS, *context_.weights_mem}});
  }
  context_.net_args.emplace_back(net_args_map);
  context_.bn_fwd.reset(new batch_normalization_forward(*context_.fwd_pd));
#else 
  // Original v0 code.
  if (!fwdParams.training && !IS_SET(use_global_stats)) {
    ...
  } else if (IS_SET(use_global_stats)) {
    ...
  } else { 
    ...
  } 
#endif  // ENABLE_MKLDNN_V1
```

Reply:
Made these changes